Tutorials/WinForms/Classic/create-persistent-classes-and-connect-xpo-to-database.md

```diff
@@ -0,0 +1,105 @@
+        get { return fLastName; }
+        set { SetPropertyValue(nameof(LastName), ref fLastName, value); }
+    }
+    ```
+    The `SetPropertyValue` method raises the `PropertyChanged` event that is used by data-bound controls to update their display text.
+* Add the read-only `ContactName` property. XPO does not map read-only properties to database columns. Use the `FirstName` and `LastName` properties to calculate the `ContactName` property value.
+    ```csharp
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Add the [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute to the `ContactName` property. 
+  >The [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute specifies an expression that substitutes for the property name in a SQL query. The expression passed to the attribute constructor should follow the [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax). 
+    ```csharp
+    [PersistentAlias("concat(FirstName, ' ', LastName)")]
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Use the same approach to add the `Order` class with the `ProductName`(String), `OrderDate`(DateTime), and `Freight`(decimal) properties. The complete code is available here:  
+  [Order.cs](/Tutorials/WinForms/Classic/CS/DataAccess/Order.cs)  
+  [Order.vb](/Tutorials/WinForms/Classic/VB/DataAccess/Order.vb)
+* Add the `Orders` and `Customer` properties to create a [relationship](https://docs.devexpress.com/XPO/2041/concepts/relationships-between-objects) between the **Customer** and **Order** classes.
+    ```csharp
+    // Customer.cs
+    [Association("CustomerOrders")]
+    public XPCollection<Order> Orders {
+        get { return GetCollection<Order>(nameof(Orders)); }
+    }
+    
+    // Order.cs
+    private Customer fCustomer;
+    [Association("CustomerOrders")]
+    public Customer Customer {
+        get { return fCustomer; }
+        set { SetPropertyValue(nameof(Customer), ref fCustomer, value); }
+    }
+    ```
+* Add a reference to the **System.Configuration** assembly. This assembly contains classes used to access connection string settings from the application's configuration file [Connection Strings and Configuration Files](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings-and-configuration-files).
+* Add the *ConnectionHelper.cs* file, copy code from the example ([ConnectionHelper.cs](/Tutorials/WinForms/Classic/CS/DataAccess/ConnectionHelper.cs)), and paste it to a new file. 
+  >This code can be used without modifications in a real application. Refer to the following article for additional details: [Connecting to a Data Store](https://docs.devexpress.com/XPO/2020/feature-center/connecting-to-a-data-store).
```

Comment:
"without modifications in a real application"?

Reply:
```suggestion
  >This code can be used in an application without any changes. Refer to the [Connecting to a Data Store](https://docs.devexpress.com/XPO/2020/feature-center/connecting-to-a-data-store) article for more information.
```

Reply:
>"without modifications in a real application"?

I replaced this sentence with "This code initializes the [Data Access Layer](https://docs.devexpress.com/XPO/2121/Feature-Center/Connecting-to-a-Data-Store/Data-Access-Layer)."

2a11ebbba1b3f716e02f7991650a8a490e88f3dc